### ISSUE TITLE
onImageRemovedCallback not called when Image is Removed with a keystroke(backspace)

### ISSUE THREAD

**Issue by Uche01** [singerdmx/flutter-quill#2113]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/singerdmx/flutter-quill/issues)

### The question

I am trying to delete an image from file system when an image is removed from the editor. Overriding the callback `onImageRemovedCallback` appears to work only when image is removed by clicking on the 'Remove' option on the image actions popup. However, when image is removed by clicking backspace key on the keyboard, the callback is not fired. How can I fix this?

Here is the call back code:
```
QuillEditorImageEmbedConfigurations imageEmbedConfigurations = QuillEditorImageEmbedConfigurations(  
    onImageRemovedCallback: (imageUrl) async {
      final file = File(imageUrl);
      if (await file.exists()) {
        await file.delete();
        print('File deleted successfully');
      } else {
        print('File does not exist');
      }
    },
  );

```

**Comment by CatHood0** [comment]
You can create a custom `Rule` that extends from `DeleteRule` and verify if the content that will be removed is an image (by now is the solution that i can think)

**Comment by CatHood0** [comment]
By now we don't have plan to add this. Probably in future releases this will be fixed (at this moment we need to take more attention to issues that are affecting to the perfomance and the stability of the package).

**Comment by EchoEllet** [comment]
There are similar issues related to this, will reopen it since I do plan on a solution when I have the time, should not make changes directly to `flutter_quill` for this fix.

> at this moment we need to take more attention to issues that are affecting to the perfomance and the stability of the package

This is why a fix will take a while before introducing it.

**Comment by EchoEllet** [comment]
A workaround is to use [`onKeyPressed`](https://github.com/singerdmx/flutter-quill/blob/c0e624a0c3f1a417edbd61d5943461686433825b/lib/src/editor/config/editor_config.dart#L140-L167) to prevent the user from deleting the image embed using the backspace.

> I am trying to delete an image from file system when an image is removed from the editor. Overriding the callback onImageRemovedCallback appears to work only when image is removed by clicking on the 'Remove' option on the image actions popup.

I'm not sure if it should be called, the editor supports the history feature with undo and redo and they can undo the change, which will restore the image and if you delete it, it will cause an error (file not found).

**Comment by CatHood0** [comment]
> A workaround is to use onKeyPressed

The problem with this is that it would only be a partial fix for desktop or web (desktop too). Since, at least on mobile, these events won't be called. Although it's better than nothing. Trying to detect the key using the TextInputClient could be too complicated (this would be the only way to allow these events to be called on all platforms with what we have).

**Comment by CatHood0** [comment]
> I'm not sure if it should be called. The editor supports the history feature with undo and redo, and they can undo the change, which will restore the image. If you delete it, it will cause a "file not found" error.

True, I've done something like that before. But since undo/redo operations are supported, I opted to implement a function that identifies the embeds that were deleted and then deletes them from the server/storage (all of this once it's confirmed that we're no longer modifying the document). 

If the OP follows what I chose to do, he/she should take into account this method of dart_quill_delta_simplify: [getAllEmbeds](https://github.com/CatHood0/dart-quill-delta-simplify/blob/18f7171e701dced8d63e0350a773093c630e689f/lib/src/extensions/query_delta_ext.dart#L69-L72).

As a quick example of this:

```dart
final List<Delta> oldEmbedsState = oldDelta.getAllEmbeds().map((DeltaRangeResult result) => result.delta); // there deltas probably will have only 1 operation that contains the embed part
final List<Delta> newEmbedsState = newDelta.getAllEmbeds().map((DeltaRangeResult result) => result.delta); // there deltas probably will have only 1 operation that contains the embed part

// ... at this point, you can make your comparation about what was removed or updated (`DeltaRangeResult` contains the offset where the embed match is positioned)
```
